core/src/data_sources/data_source.rs

```diff
@@ -800,194 +802,81 @@ impl DataSource {
-                            None => (),
-                        };
-                    }
-                    None => (),
-                };
-
-                match f.parents {
-                    Some(parents) => {
-                        match parents.is_in.clone() {
-                            Some(v) => must_filter.push(qdrant_match_field_condition("parents", v)),
-                            None => (),
-                        };
-                        match parents.is_not.clone() {
-                            Some(v) => {
-                                must_not_filter.push(qdrant_match_field_condition("parents", v))
-                            }
-                            None => (),
-                        };
-                    }
-                    None => (),
-                };
-
-                match f.timestamp {
-                    Some(timestamp) => {
-                        match timestamp.gt.clone() {
-                            Some(v) => must_filter.push(
-                                qdrant::FieldCondition {
-                                    key: "timestamp".to_string(),
-                                    range: Some(qdrant::Range {
-                                        gte: Some(v as f64),
-                                        ..Default::default()
-                                    }),
-                                    ..Default::default()
-                                }
-                                .into(),
-                            ),
-                            None => (),
-                        };
-                        match timestamp.lt.clone() {
-                            Some(v) => must_filter.push(
-                                qdrant::FieldCondition {
-                                    key: "timestamp".to_string(),
-                                    range: Some(qdrant::Range {
-                                        lte: Some(v as f64),
-                                        ..Default::default()
-                                    }),
-                                    ..Default::default()
-                                }
-                                .into(),
-                            ),
-                            None => (),
-                        };
-                    }
-                    None => (),
-                };
-
-                Some(qdrant::Filter {
-                    must: must_filter,
-                    must_not: must_not_filter,
-                    ..Default::default()
-                })
+        let chunks = match query.is_none() {
+            true => {
+                let store = store.clone();
+                if !target_document_tokens.is_none() {
+                    Err(anyhow!(
+                        "target_document_tokens is only supported with query"
+                    ))?;
+                }
+                self.retrieve_chunks_without_query(store, &qdrant_client, top_k, &filter)
```

Comment:
Let's add a DSSTAT like log?

Reply:
Would you add inside of the helper or around ?

Reply:
outside since the start time is defined here?

Reply:
Done